lib/data-types.js

```diff
@@ -526,6 +526,36 @@ ARRAY.is = function(obj, type) {
   return obj instanceof ARRAY && obj.type instanceof type;
 };
 
+var GEOMETRY = function(type, srid) {
+  var options = _.isPlainObject(type) ? type : {
+    type: type,
+    srid: srid
+  };
+
+  if (!(this instanceof GEOMETRY)) return new GEOMETRY(options);
+
+  this.options = options;
+  this.type = options.type;
+  this.srid = options.srid;
+};
+util.inherits(GEOMETRY, ABSTRACT);
+
+GEOMETRY.prototype.key = GEOMETRY.key = 'GEOMETRY';
```

Comment:
My original though is that maybe the parsing should be done in the GEOMETRY
data type object. That way the code is in an obvious place. I'll check out
the range object.

Mike Ringrose

On Fri, May 29, 2015, 5:04 PM Jan Aagaard Meier <notifications@github.com>
wrote:

> In lib/data-types.js
> <https://github.com/sequelize/sequelize/pull/3828#discussion_r31364686>:
>
> > @@ -526,6 +526,36 @@ ARRAY.is = function(obj, type) {
> >    return obj instanceof ARRAY && obj.type instanceof type;
> >  };
> >
> > +var GEOMETRY = function(type, srid) {
> > +  var options = _.isPlainObject(type) ? type : {
> > +    type: type,
> > +    srid: srid
> > +  };
> > +
> > +  if (!(this instanceof GEOMETRY)) return new GEOMETRY(options);
> > +
> > +  this.options = options;
> > +  this.type = options.type;
> > +  this.srid = options.srid;
> > +};
> > +util.inherits(GEOMETRY, ABSTRACT);
> > +
> > +GEOMETRY.prototype.key = GEOMETRY.key = 'GEOMETRY';
>
> These type defs should be moved to postgres datatypes, and then the
> general datatypes should be WKT
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sequelize/sequelize/pull/3828/files#r31364686>.
>